diff --git a/avidemux/osxInstaller/CMakeLists.txt b/avidemux/osxInstaller/CMakeLists.txt
index fd44d0270..8d1519d8c 100644
--- a/avidemux/osxInstaller/CMakeLists.txt
+++ b/avidemux/osxInstaller/CMakeLists.txt
@@ -57,8 +57,8 @@ ENDIF()
 #
 MESSAGE(STATUS "* Creating Skeleton *")
 # Quick Hack...
-#FILE(COPY "${CMAKE_CURRENT_SOURCE_DIR}/../qt4/ADM_userInterfaces/ADM_gui/pics/avidemux.icns"
-#  DESTINATION "${ORIGINAL}/Contents/Resources/avidemux.icns")
+FILE(COPY "${CMAKE_CURRENT_SOURCE_DIR}/../qt4/cmake/osx/avidemux.icns"
+  DESTINATION "${ORIGINAL}/Contents/Resources/")
 FILE(COPY "${ORIGINAL}/Contents/MacOS/Avidemux${AVIDEMUX_MAJOR_MINOR}" DESTINATION "${BUNDLE}/Contents/MacOS/")
 # Copy jobs and cli too
 FILE(COPY "${ORIGINAL}/Contents/Resources/bin/avidemux_cli"          DESTINATION "${BUNDLE}/Contents/MacOS/")
diff --git a/bootStrap.bash b/bootStrap.bash
index 68055cdfa..7ebe8a2f1 100644
--- a/bootStrap.bash
+++ b/bootStrap.bash
@@ -98,7 +98,13 @@ Process() {
   $ADATE
   echo "   $BASE:Install started..."
   # we need the make install so that other packcges can be built against this one
-  DESTDIR="${FAKEROOT_DIR}" ${MAKER} install >&/tmp/logInstall$BASE || fail "install faied, see /tmp/logInstall$BASE"
+  if DESTDIR="$FAKEROOT_DIR" ${MAKER} -n install; then
+    DESTDIR="${FAKEROOT_DIR}" ${MAKER} install >&/tmp/logInstall$BASE || fail "install faied, see /tmp/logInstall$BASE"
+  else
+    DESTDIR="$FAKEROOT_DIR" ${MAKER} preinstall
+    cmake -E cmake_echo_color "--switch=$color" --cyan "Install the project..."
+    DESTDIR="${FAKEROOT_DIR}" ${MAKER} preinstall >&/tmp/logInstall$BASE || fail "install faied, see /tmp/logInstall$BASE"
+  fi
   popd >/dev/null
   $ADATE
   echo "   Done "
diff --git a/bootStrapMacOS_Monterey.arm64.sh b/bootStrapMacOS_Monterey.arm64.sh
index 3747c5a72..09e2da26a 100644
--- a/bootStrapMacOS_Monterey.arm64.sh
+++ b/bootStrapMacOS_Monterey.arm64.sh
@@ -67,7 +67,13 @@ Process()
     else
         echo "** installing to $PREFIX **"
     fi
-    DESTDIR="$FAKEROOT_DIR" make install || fail install
+    if DESTDIR="$FAKEROOT_DIR" make -n install ; then
+        DESTDIR="$FAKEROOT_DIR" make install || fail install
+    else
+        DESTDIR="$FAKEROOT_DIR" make preinstall
+        cmake -E cmake_echo_color "--switch=$color" --cyan "Install the project..."
+        DESTDIR="$FAKEROOT_DIR" cmake -P cmake_install.cmake || fail install
+    fi
     popd > /dev/null
 }
 printModule()
diff --git a/bootStrapOsx_Catalina.bash b/bootStrapOsx_Catalina.bash
index 2eda71dcf..9dc9f6224 100644
--- a/bootStrapOsx_Catalina.bash
+++ b/bootStrapOsx_Catalina.bash
@@ -68,7 +68,13 @@ Process()
     else
         echo "** installing to $PREFIX **"
     fi
-    DESTDIR="$FAKEROOT_DIR" make install || fail install
+    if DESTDIR="$FAKEROOT_DIR" make -n install ; then
+        DESTDIR="$FAKEROOT_DIR" make install || fail install
+    else
+        DESTDIR="$FAKEROOT_DIR" make preinstall
+        cmake -E cmake_echo_color "--switch=$color" --cyan "Install the project..."
+        DESTDIR="$FAKEROOT_DIR" cmake -P cmake_install.cmake || fail install
+    fi
     popd > /dev/null
 }
 printModule()
diff --git a/macos_brew_setup.txt b/macos_brew_setup.txt
index f9a76ceb3..e3b78d27f 100644
--- a/macos_brew_setup.txt
+++ b/macos_brew_setup.txt
@@ -1,2 +1,5 @@
+#!/bin/bash
+brew install cmake pkg-config nasm yasm qt xvid x264 x265 libvpx aom opus fdk-aac lame libass mp4v2 a52dec
 brew install make cmake yasm  pkg-config bzip2 zlib patchutils  sqlite3 libogg libvorbis opus libopusenc lame two-lame
 brew install aom libvpx libass x264 x265 qt faac faad2 xvid
+brew install vapoursynth aften libdca opencore-amr ffmpeg llvm libomp
