diff --git a/avidemux/osxInstaller/macos-adhoc-sign-installed-app.sh b/avidemux/osxInstaller/macos-adhoc-sign-installed-app.sh
index 236a581f8..dfae4d901 100755
--- a/avidemux/osxInstaller/macos-adhoc-sign-installed-app.sh
+++ b/avidemux/osxInstaller/macos-adhoc-sign-installed-app.sh
@@ -9,7 +9,7 @@ sign()
     codesign --force -s - $1 || fail "Error ad-hoc signing $1"
 }
 
-BUNDLE_CONTENT="/Applications/Avidemux-2.8.2.app/Contents"
+BUNDLE_CONTENT="$GITHUB_WORKSPACE/avidemux2/Avidemux2.8.app/Contents"
 
 if [ ! -d ${BUNDLE_CONTENT} ]
 then
diff --git a/bootStrapMacOS_Monterey.arm64.sh b/bootStrapMacOS_Monterey.arm64.sh
index 3747c5a72..23722b5af 100644
--- a/bootStrapMacOS_Monterey.arm64.sh
+++ b/bootStrapMacOS_Monterey.arm64.sh
@@ -67,7 +67,13 @@ Process()
     else
         echo "** installing to $PREFIX **"
     fi
-    DESTDIR="$FAKEROOT_DIR" make install || fail install
+    if DESTDIR="$FAKEROOT_DIR" make -n install ; then
+        DESTDIR="$FAKEROOT_DIR" make install || fail install
+    else
+        DESTDIR="$FAKEROOT_DIR" make preinstall
+        cmake -E cmake_echo_color "--switch=$color" --cyan "Install the project..."
+        DESTDIR="$FAKEROOT_DIR" cmake -P cmake_install.cmake || fail install
+    fi
     popd > /dev/null
 }
 printModule()
